### PR TITLE
Confirmations implementation

### PR DESCRIPTION
**Pull request by NejcZdovc**
fixes https://github.com/brave-intl/bat-native-confirmations/issues/50

### CODE REVIEW

File: src/bat/confirmations/issuer_info.cc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/confirmations/issuer_info.h"
+
+namespace confirmations {
+
+IssuerInfo::IssuerInfo() :
+    name(""),
```
Comment (emerick) [inline src/bat/confirmations/issuer_info.cc]: Nitpick: don't really need to do this, as std::string initializes to empty by default.

File: src/confirmations_impl.h [hunk 2]
```diff
@@ -2,98 +2,199 @@
...
+
+  /////////////////////////////////////////////////////////////////////////////
+
+  void SetConfirmationsStatus();
+
+  std::string GetServerUrl();
+  int GetServerPort();
+
+  void Step1StoreTheServersConfirmationsPublicKeyAndGenerator(
+      std::string confirmations_GH_pair,
```
Comment (emerick) [inline src/confirmations_impl.h]: Should this be const?

File: src/confirmations_impl.h [hunk 3]
```diff
@@ -2,98 +2,199 @@
...
+      std::vector<std::string> bat_names,
+      std::vector<std::string> bat_keys);
+
+  void Step2RefillConfirmationsIfNecessary(
+      std::string real_wallet_address,
```
Comment (emerick) [inline src/confirmations_impl.h]: Seems like these params should be const.

File: src/confirmations_impl.h [hunk 4]
```diff
@@ -2,98 +2,199 @@
...
+
+  void SaveState();
+  void OnStateSaved(const Result result);
+  void LoadState();
+  void OnStateLoaded(const Result result, const std::string& json);
+  void ResetState();
+  void OnStateReset(const Result result);
+
+  std::string ToJSON();
+  bool FromJSON(std::string json);
```
Comment (emerick) [inline src/confirmations_impl.h]: Should be const params. We should probably check all of the function signatures when fixing this.

File: src/confirmations_impl.h [hunk 5]
```diff
@@ -2,98 +2,199 @@
...
+
+  void VectorConcat(
```
Comment (emerick) [inline src/confirmations_impl.h]: Seems weird that these are pointers...

File: src/confirmations_impl.h (hunk 2 above)
Comment (tmancey) [reply src/confirmations_impl.h]: This library is not const correct, but I did not want to change original code, so ok to fix later?

File: src/confirmations_impl.h (hunk 3 above)
Comment (tmancey) [reply src/confirmations_impl.h]: This library is not const correct, but I did not want to change original code, so ok to fix later?

File: src/confirmations_impl.h (hunk 4 above)
Comment (tmancey) [reply src/confirmations_impl.h]: This library is not const correct, but I did not want to change original code, so ok to fix later?

File: src/confirmations_impl.h (hunk 5 above)
Comment (emerick) [reply src/confirmations_impl.h]: Yup.

File: src/confirmations_impl.h (hunk 3 above)
Comment (emerick) [reply src/confirmations_impl.h]: Yup.

**Pull request merged**